Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/boosterinfodetails/BoosterInfoDetailsViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package de.rki.coronawarnapp.covidcertificate.boosterinfodetails
+
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.BoosterNotification
+import de.rki.coronawarnapp.ccl.dccwalletinfo.text.textResource
+import de.rki.coronawarnapp.ccl.dccwalletinfo.text.urlResource
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesProvider
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.mapNotNull
+import timber.log.Timber
+
+class BoosterInfoDetailsViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val personCertificatesProvider: PersonCertificatesProvider,
+    @AppScope private val appScope: CoroutineScope,
+    @Assisted private val personIdentifierCode: String
+) : CWAViewModel(dispatcherProvider) {
+
+    private val uiStateFlow = personCertificatesProvider.personCertificates.mapNotNull { certificateSet ->
+        UiState(
+            certificateSet.first { it.personIdentifier?.codeSHA256 == personIdentifierCode }
+                .dccWalletInfoWrapper.dccWalletInfo.boosterNotification
+        )
+    }.catch { error ->
+        Timber.d(error, "No person found for $personIdentifierCode")
+    }
+    val uiState = uiStateFlow.asLiveData2()
+
+    data class UiState(
+        val boosterNotification: BoosterNotification
+    ) {
+        val titleText by textResource(boosterNotification.titleText)
+        val subtitleText by textResource(boosterNotification.subtitleText)
+        val longText by textResource(boosterNotification.longText)
+        val faqUrl by urlResource(boosterNotification.faqAnchor)
+    }
+
+    fun onViewed() {
+        // TODO: add correct call to change notification status to read/viewed
```

Comment:
which subtask covers this TODO?

Reply:
[Subtask](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11592)